enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,156 @@
+---
+title: openstack-baremetal-compute
+authors:
+  - "@pierreprinetti"
+reviewers:
+  - "@openshift/team-openstack-cloud-provider"
+approvers:
+  - TBD
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: provisional
+
+---
+
+# Openstack Baremetal Compute
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenStack's [Nova][openstack-nova] service can be configured to provision
+baremetal machines as OpenShift Compute Nodes.
```

Comment:
No need to artificially limit ourselves to OpenShift Compute nodes, unless there is a technical reason why OpenStack isn't capable of provisioning OpenShift Control Plane nodes of course.

Reply:
You bring an interesting point here.

I tend to be conservative and tightly tailor the enhancement upon the user story we are given (i.e. Compute for performance): I am afraid of introducing any complexity which is not strictly related to the (known) user needs.

However, you have far better insight then I have on the matter, and you might already know that (either):
* flexibility is a requirement; OR
* extending the enhancement adds tolerable complexity.

So, do we want to limit our support case to Compute nodes, with the option to add more to it later, or can we confidently generalise already?

Reply:
From a mechanics standpoint with-in openstack, there is no difference, Your passing ignition data along. 

From what I understand of openshift, it is beyond the scope of the cloud provider to solely spin up new controllers, so by trying to pull that in as well, it feels very strongly of scope creep.

I recommend keeping things specific to the user story such that the worker node count can be set and things can be scaled through the cloud provider.